Do not encode JSON objects again

XVincentX opened the pull request:
This PR makes sure to check out what kind of parameter we're treating when creating a message body.
If it's an object, then we can serialise it as JSON, but if it's not, we should just forward it to the `Asset` object. If not, we will double encode it and a `JSON.parse` conversion will not give the desired result.

XVincentX commented:
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;X9c4SgAK&#x2F;98-fix-json-examples"><img src="https:&#x2F;&#x2F;github.trello.services&#x2F;images&#x2F;trello-icon.png" width="12" height="12"> Fix JSON examples</a>

smizell commented on .editorconfig:
```diff
@@ -0,0 +1,11 @@
+insert_final_newline = true
+indent_style = space
+indent_size = 2
+trim_trailing_whitespace = true
```
A [Rider](https://en.wikipedia.org/wiki/Rider_(legislation)) :)

Looks good to me though.

kylef commented on src/adapter.js:
```diff
@@ -695,7 +695,13 @@ export function parse({minim, source, generateSourceMap}, done) {
-              const bodyAsset = new Asset(JSON.stringify(responseBody, null, 2));
+              let formattedResponseBody = responseBody;
+
+              if (typeof(responseBody) === 'object') {
```
Would array's of objects also need to be stringified too?

smizell replied:
Does this work? Can't valid JSON be any primitive value, from string, to array, to object? I will need to look at the Swagger spec, but I'm thinking we'll need to stringify anything other than a string (assuming a string is either a primitive string or JSON-stringified).

XVincentX replied:
Right, I will revert the condition then-

smizell commented on test/fixtures/refract/payload-as-string.json:
```diff
@@ -0,0 +1,208 @@
+        "classes": [
+          "api"
+        ],
+        "title": "Not formated example"
```
Nit: formatted misspelled

XVincentX replied:
Thanks, will fix

smizell commented on test/fixtures/refract/x-summary-and-description.json:
```diff
@@ -52,7 +52,9 @@
-        "classes": ["api"]
+        "classes": [
+          "api"
+        ]
```
Specific reason for this?

pksunkara replied:
It's how the generated JSONs are serialized.

smizell replied:
True, but I'm not sure why this is a change to this PR as it seems unrelated. Is it making things consistent? Or was this making tests fail?

XVincentX replied:
Well it was regenerated by `npm` test command. Will fix!

pksunkara replied:
I don't think this needs to be fixed.

smizell replied:
No problem to fix if it was auto generated. Was just curious why it was there.

smizell replied:
I mean, you don't need to fix it.

XVincentX commented on test/fixtures/swagger/payload-as-string.yaml:
```diff
@@ -0,0 +1,47 @@
+              }
+
+        default:
+          description: Error
```
True that, will fix.

XVincentX commented:
@smizell Thanks for the review. I've addressed all the comments; changed the condition, added new lines where needed and stripped down the fixture to the minimal possible one. Please have a new look, thanks!
